### PR TITLE
libraries: implement three new types of notification

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Implements three new notifications: booking, request and transit_notice.
* Implements a new field communication language.
* Customizes message by type of notifications.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- Implements 3 new notifications

## Dependencies

My PR depends PR(s):

* https://github.com/rero/rero-ils/pull/1914

## How to test?

- Check notifications form

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — CHANGES_REQUESTED** [review]
A verb is missing in the commit message title.
The three kind of notifications could be listed in the commit message.

File: projects/admin/src/app/record/custom-editor/libraries/library.component.html [hunk 1]
```diff
@@ -176,10 +176,13 @@ <h1 translate>Library</h1>
                     translate>Email</label>
                   <div class="col-md-5">
                     <input formControlName="email" class="form-control" [id]="setting.controls.type.value + '-email'">
-                    <div class="text-secondary" translate>
+                    <div *ngIf="setting.controls.type.value|notificationType:'patron'" class="text-secondary" translate>
```
Comment (zannkukai) [inline projects/admin/src/app/record/custom-editor/libraries/library.component.html:179]: add spaces before and after the '|'

File: projects/admin/src/app/record/custom-editor/libraries/library.component.html [hunk 2]
```diff
@@ -176,10 +176,13 @@ <h1 translate>Library</h1>
...
                       The notifications for patrons without e-mail are sent to this address. No notification is sent if
                       the field is empty.
                     </div>
+                    <div *ngIf="setting.controls.type.value|notificationType:'library'" class="text-secondary" translate>
```
Comment (zannkukai) [inline projects/admin/src/app/record/custom-editor/libraries/library.component.html:183]: same here

File: projects/admin/src/app/record/custom-editor/libraries/pipe/notificationType.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      'due_soon',
+      'overdue',
+      'recall'
+    ],
+    library: [
+      'booking',
+      'request',
+      'transit_notice'
+    ]
+  };
```
Comment (zannkukai) [inline projects/admin/src/app/record/custom-editor/libraries/pipe/notificationType.pipe.ts:36]: For me, these values should be deported into a 'Notification class model' outside of the pipe.
In few months, it shouldn't be easy to remember than this config is into this Pipe.

**Review by iGormilhit — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/library-detail-view/library-detail-view.component.html [hunk 4]
```diff
@@ -40,6 +40,13 @@ <h1>{{ record.metadata.name }}</h1>
     <dd class="col-sm-7 col-md-8 mb-0">
       {{ record.metadata.email }}
     </dd>
+    <!-- COMMUNICATION LANGUAGE -->
+    <dt class="col-sm-3 offset-sm-2 offset-md-0 label-title" *ngIf="record.metadata.communication_language" translate>
+      Communication language
+    </dt>
+    <dd class="col-sm-7 col-md-8 mb-0">
+      {{ ('lang_' + record.metadata.communication_language) | translate | ucfirst }}
+    </dd>
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/library-detail-view/library-detail-view.component.html:49]: same remarks, especially here where we use "offset-sm" and "offset-md"... coud be change into simpliest `col-3 offset-2` and `col-7`

**Pull request merged**